simulator/build.gradle.kts

```diff
@@ -91,3 +91,29 @@ tasks.register<Copy>("untarTestBlockStream") {
 tasks.named("processResources") { dependsOn(tasks.named("untarTestBlockStream")) }
 
 tasks.named("sourcesJar") { dependsOn(tasks.named("untarTestBlockStream")) }
+
+// Task to prepare and run Docker container
+tasks.register("startSimulatorDockerContainers") {
+    description = "Prepare and run the simulator containers in publisher and consumer mode"
+    group = "docker"
+
+    doLast {
+        // Make the prepare and cleanup scripts executable
+        exec { commandLine("chmod", "+x", "docker/prepare-docker.sh", "docker/cleanup-docker.sh") }
+
+        // Run the prepare script
+        exec { commandLine("./docker/prepare-docker.sh") }
+
+        // Change to docker directory and run docker-compose
+        exec {
+            workingDir("docker")
+            commandLine("sh", "-c", "docker compose -p simulator up --build -d")
+        }
+
+        // Run cleanup script
+        exec { commandLine("./docker/cleanup-docker.sh") }
```

Comment:
Why ? I don't want to manually delete some files everytime.

Reply:
If we never leave the volatile build root and we generate everything that is temporary and needs to be scrapped there, then we really do not need to delete any files, as the build root is not under the management of the source control and with the next run of gradle's clean task, all files will be scrapped.

Reply:
Yeah, but i'm currently using them from docker folder, will modify to try to take them without copying.

Reply:
The point here is to copy the docker folder into the build root of the subproject. The docker folder must contain files which when used with the build root as a working directory, they will not need to go out of it to supply themselves with anything that is external to the build root. In this way we ensure that the builds are only restricted to the build root (which should be the case).

Reply:
+1 @ata-nas The build is creating/copying files already, so it's entirely reasonable to copy docker definitions into the build folder and write any changes there only, so that the build folder is the only thing modified during a build.

This makes system audits easier and also makes it easier to create repeatable builds because we always start from a clean base.

Reply:
Changed to copy and work in build root.